server.js

```diff
@@ -1,30 +1,58 @@
-// Log memory usage before doing GC. Seems to blow up the file limit sooner than without.
-var gcLog = function () {
-  console.log(new Date().toISOString() + JSON.stringify(process.memoryUsage()));
-  gc();
-};
+var fs = require('fs');
+var ws = require('ws');
+var http = require('http');
+var randomBytes = require('crypto').randomBytes;
 
-// Tell original client how many clients are connected
-var reporter = function () {
-  if (this.clients[0]) this.clients[0].send(this.clients.length.toString());
-};
+var html = fs.readFileSync('./index.html');
+var browsers = [];
 
-// Garbage collect every 30 seconds
-setInterval(gc, 30 * 1000);
+var www = http.createServer(function (req, res) {
+  res.writeHead(200, { 'Content-Type': 'text/html' });
+  res.end(html);
+}).listen(process.env.PORT || 5000);
 
-// Create the WS server
-new (require('ws').Server)({
+var wss = new ws.Server({
+  server: www
+}).on('connection', function (socket) {
+  socket.on('message', onmessage);
+  socket.on('close', logSocketCount);
+  logSocketCount();
+});
 
-  // Attach it to an HTTP server that serves a tiny web client
-  server: require('http').createServer(function (req, res) {
+// Log memory usage before doing GC. Seems to blow up the file limit sooner than without.
+setInterval(logMemoryUsage, 30 * 1000);
 
-    res.writeHead(200, { 'Content-Type': 'text/html' });
-    res.end('<script>new WebSocket("ws://"+location.host).onmessage=function(msg){document.body.innerText=msg.data;};</script>');
+function logMemoryUsage () {
+  console.log(new Date().toISOString() + JSON.stringify(process.memoryUsage()));
+  gc();
+}
 
-  }).listen(process.env.PORT || 5000)
+function getSocketCount () {
+  return wss.clients.length.toString();
+}
 
-// Call the reporter on connection and close events
-}).on('connection', function (connection) {
-  reporter.call(this);
-  connection.on('close', reporter.bind(this));
-});
+function logSocketCount () {
+  var n = getSocketCount();
+  browsers.forEach(function (b) {
+    b.readyState === 1 && b.send(n);
+  });
```

Comment:
I guess the second link should have been to this, since we're in node-land here:

https://github.com/websockets/ws/blob/master/lib/WebSocket.js#L212